### PR TITLE
Enable weights for MG similarity algorithms 

### PR DESCRIPTION
**Pull request by jnke2016**
This is a follow up PR to #3828 which enabled weighted for the python SG similarity algorithms. 
This PR also updates the tests, docstrings and remove experimental calls

### CODE REVIEW

File: python/cugraph/cugraph/tests/link_prediction/test_overlap_mg.py [hunk 1]
```diff
@@ -72,7 +76,10 @@ def input_expected_output(input_combo):
     input_data_path = input_combo["graph_file"]
     directed = input_combo["directed"]
     has_vertex_pair = input_combo["has_vertex_pair"]
-    G = utils.generate_cugraph_graph_from_file(input_data_path, directed=directed)
+    is_weighted = input_combo["is_weighted"]
```
Comment (naimnv) [inline python/cugraph/cugraph/tests/link_prediction/test_overlap_mg.py:79]: I wonder, wouldn't it be cleaner to have these column names in global variable declared at the beginning of the files?

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Pull request merged**